homeassistant/components/telegram_bot/polling.py

```diff
@@ -22,10 +22,11 @@ async def async_setup_platform(hass, bot, config):
     return True
 
 
-def process_error(update: Update, context: CallbackContext):
+def process_error(update: Update, context: CallbackContext) -> None:
     """Telegram bot error handler."""
     try:
-        raise context.error
+        if context.error:
```

Comment:
I'm having a little trouble understanding the semantics of the telegram library error handler callback and what it's trying to accomplish.  It appears to ignore some types of exceptions, log on some type, then it also lets some just raise through back up into the callback handler.  It seems like the case where this is `None` is a shouldn't happen case and previously if it were being hit it would raise a `TypeError`.  

Anyway, just looking for a little assurance for why this is generally OK, or not important to understand at this level of detail :)